scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,450 @@
+        if ret_code != 0:
+            print("Failed to get ha cluster status")
+            res_string = ""
+        self._ufm_ha_status_string = res_string
+
+    def _get_status_value_from_ha_cluster_status(self, keyword: str):
+        for line in self._ufm_ha_status_string.splitlines():
+            line = line.strip()
+            if line.startswith(keyword):
+                val = line.split(":", 1)[-1].strip().lower()
+                val = (
+                    val.replace("[", "")
+                    .replace("]", "")
+                    .replace("(", "")
+                    .replace(")", "")
+                )
+                return val
+        return None
+
+    def check_drbd_connectivity(self):
+        if not self._ufm_ha_status_string:
+            return False
+        drbd_connectivity_status = self._get_status_value_from_ha_cluster_status(
+            "DRBD_CONNECTIVITY"
+        )
+        if drbd_connectivity_status != "connected":
+            print(
+                f"DRBD CONNECTIVITY status is {drbd_connectivity_status} and not Connected"
+            )
+            return False
+        print("DRBD CONNECTIVITY status is ok - Connected")
+        return True
+
+    def check_drdb_role(self):
+        if not self._ufm_ha_status_string:
+            return False
+        drdb_resource_status = self._get_status_value_from_ha_cluster_status(
+            "DRBD_ROLE"
+        )
+        if drdb_resource_status != "secondary":
+            print(f"DRDB ROLE status is {drdb_resource_status} and not Secondary")
+            return False
+        print("DRDB ROLE status is ok - Secondary")
+        return True
+
+    def check_drbd_disk_state(self):
+        if not self._ufm_ha_status_string:
+            return False
+        drbd_disk_state = self._get_status_value_from_ha_cluster_status("DISK_STATE")
+        if drbd_disk_state != "uptodate":
+            print(f"DRBD DISK STATE status is {drbd_disk_state} and not UpToDate")
+            return False
+        print("DRBD DISK STATE status is ok - UpToDate")
+        return True
+
+    def run_advanced_checks(self):
+        result = True
+        self.set_ufm_ha_cluster_status_string()
+        checks = [
+            self._check_pcs_status,
+            self._check_corosync_rings_status,
+            self.check_drdb_role,
+            self.check_drbd_connectivity,
+            self.check_drbd_disk_state,
+        ]
+        for check in checks:
+            if not check():
+                result = False
+        check_with_arg = [
+            (self.check_if_service_is_active, "corosync"),
+            (self.check_if_service_is_active, "pacemaker"),
+            (self.check_if_service_is_active, "pcsd"),
+        ]
+        for check, arg in check_with_arg:
+            if not check(arg):
+                result = False
+        return result
+
+
+def main(args):
+    standby_node_checker = StandbyNodeHealthChecker(
```

Comment:
I recommend to wrap the main with try...except for unexpected errors + allow the user to press CTRL+C to quit the test

Reply:
Done